packages/salesforce-adapter/src/fetch.ts

```diff
@@ -188,15 +196,20 @@ const getFullName = (obj: FileProperties, addNamespacePrefixToFullName?: boolean
     }
     return obj.fullName
   }
+
   if (obj.fullName.startsWith(namePrefix)) {
     return obj.fullName
   }
   if (addNamespacePrefixToFullName) {
-    // In some cases, obj.fullName does not contain the namespace prefix even though
-    // obj.namespacePrefix is defined. In these cases, we want to add the prefix manually
+  // In some cases, obj.fullName does not contain the namespace prefix even though
+  // obj.namespacePrefix is defined. In these cases, we want to add the prefix manually
+    if (obj.fullName.includes(API_NAME_SEPARATOR)) {
+    // Case for API name with an object <Object>.X where it should be <Object>.<namespace>__X
+      return obj.fullName.replace(API_NAME_SEPARATOR, `${API_NAME_SEPARATOR}${namePrefix}`)
```

Comment:
Nit:
While this is correct, this logic seems vague for me, and since this value is **EXTREMELY** important I'd prefer to be explicit here.
```suggestion
      [parentName, instanceName] = obj.fullName.split(API_NAME_SEPARATOR)
      return `${parentName}.${namePrefix}${instanceName}`
```

Reply:
I really think we should not make this change "blindly"
we should stick with logging first to make sure we don't ruin other scenarios that currently work

Reply:
@ori-moisis There won't be any change unless the flag is off. Until then, there will only be logging on the failed scenario. Do you mean to split the flag up so that each sub-scenario is logged?